packages/wire/src/queue.ts

```diff
@@ -27,19 +28,83 @@ export class AsyncQueue {
   } | null = null;
   constructor() {
     this.socket = new net.Socket({});
-    this.socket.on('data', (buffer: Buffer) => {
-      debug('received %o bytes', buffer.length);
-      this.queue.push(buffer);
-      this.processQueue();
-    });
   }
-  public connect(passedOptions: { port: number; host: string }): Promise<void> {
+  public connect(passedOptions: {
+    port: number;
+    host: string;
+    ssl?: tls.ConnectionOptions | boolean;
+  }): Promise<void> {
+    const { ssl, ...connectOptions } = passedOptions;
+    const sslEnabled = ssl === true || ssl != null;
+
+    const attachDataListener = () => {
+      this.socket.on('data', (buffer: Buffer) => {
+        debug('received %o bytes', buffer.length);
+        this.queue.push(buffer);
+        this.processQueue();
+      });
+    };
+
     return new Promise((resolve) => {
       this.socket.on('connect', () => {
         debug('socket connected');
-        resolve();
+
+        if (sslEnabled) {
+          const requestSSLMessage = Buffer.allocUnsafe(8);
+          requestSSLMessage.writeInt32BE(8, 0);
+          requestSSLMessage.writeInt32BE(80877103, 4);
```

Comment:
Can this message be moved into the messages modules?
https://github.com/adelsz/pgtyped/blob/master/packages/wire/src/messages.ts

Reply:
Yea I was going to go about doing that, but unsure how to convert it to the patterns array.

I've tried doing patterns: [ int32(8), int(80877103) ], but it adds some extra bits of info in the beginning.

Reply:
The first 4 bytes is the message length which is added automatically by the message processor, will something like this work?
```
sslRequest: {
    name: 'SSLRequestMessage',
    type: 'CLIENT',
    indicator: null,
    pattern: (data) => [
      // The SSL request code.
      int32(80877103),
    ],
  }
```

Reply:
Yup that seems to work!